cse-lib/data/src/main/java/com/farao_community/farao/cse/data/ntc/NtcUtilAdapted.java

```diff
@@ -24,7 +24,13 @@ private NtcUtilAdapted() {
         // Should not be instantiated
     }
 
-    static TNTC getTNtcFromLine(OffsetDateTime targetDateTime, TLine tLine) {
+    static Optional<TNTC> getTNtcFromLineFromNtcRedFile(OffsetDateTime targetDateTime, TLine tLine) {
+        return getTNtcFromPeriods(targetDateTime, tLine.getPeriod())
+                .stream()
+                .collect(toOptional());
+    }
+
+    static TNTC getTNtcFromLineFromYearlyNtc(OffsetDateTime targetDateTime, TLine tLine) {
```

Comment:
I am not sure this is a good idea to create two different methods that do mostly the same (except throwing exception) with names that do not really explain the difference between both and are strongly linked to the context in which the methods are used. Especially since each ot them is only used once in the application.

I suggest coming back to a single method named `getTNtcFromLine()` which would do the job of  `getTNtcFromLineFromNtcRedFile()` (i.e. return an Optional) and put the `.orElseThrow()` instruction directly in YearlyNtcDocumentAdapted where you call the method.